compiler/rustc_target/src/target_features.rs

```diff
@@ -607,8 +613,14 @@ impl super::spec::Target {
     pub fn features_for_correct_vector_abi(&self) -> Option<&'static [(u64, &'static str)]> {
         match &*self.arch {
             "x86" | "x86_64" => Some(X86_FEATURES_FOR_CORRECT_VECTOR_ABI),
-            "aarch64" => Some(AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI),
-            // FIXME: add support for non-tier1 architectures
+            "arm" | "aarch64" => Some(AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI),
```

Comment:
It seems suspicious to treat these together here -- they generally have completely independent sets of target features.

Reply:
Fair, split it up. Also added a couple of "Tier 1 / Tier 2" comments.